modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/state.scala

```diff
@@ -0,0 +1,155 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+
+/**
+ * Flag to indicate whether the global execution is `Running` or `Waiting`.
+ */
+sealed trait Status
+
+object Status {
```

Comment:
For now, I could get away with a single status type at every level but I think we would soon have the need for different types of status at each level.